sdk/servicebus/Azure.Messaging.ServiceBus/tests/Samples/Sample15_MockingClientTypes.cs

```diff
@@ -0,0 +1,665 @@
+                    It.IsAny<CancellationToken>()))
+                .Callback<String, String, CreateRuleOptions, CancellationToken>((topic, sub, opts, ct) =>
+                {
+                    RuleProperties mockRuleProperties = ServiceBusModelFactory.RuleProperties(opts);
+                    ;
+
+                    mockRuleResponse.Setup(r => r.Value).Returns(mockRuleProperties);
+                })
+                .ReturnsAsync(mockRuleResponse.Object);
+
+            ServiceBusAdministrationClient administrationClient = mockAdministrationClient.Object;
+
+            // The rest of this snippet illustrates how to create a rule using the mocked service bus
+            // administration client above, this would be where application methods creating a rule would be
+            // called.
+
+            string ruleName = "rule";
+            CreateRuleOptions options = new(ruleName)
+            {
+                Filter = new CorrelationRuleFilter { Subject = "subject" }
+            };
+
+            string topic = "topic";
+            string subscription = "subscription";
+            RuleProperties createRule = await administrationClient.CreateRuleAsync(topic, subscription, options);
+            #endregion
+        }
+
+        [Test]
+        public async Task MockRuleManager()
+        {
+            #region Snippet:ServiceBus_MockingRuleManager
+            Mock<ServiceBusClient> mockClient = new();
+            Mock<ServiceBusRuleManager> mockRuleManager = new();
+
+            mockRuleManager
+                .Setup(rm => rm.DeleteRuleAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<CancellationToken>()))
+                .Returns(Task.CompletedTask);
+
+            mockRuleManager
+                .Setup(rm => rm.CreateRuleAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<RuleFilter>(),
+                    It.IsAny<CancellationToken>()))
+                .Returns(Task.CompletedTask);
+
+            mockClient
+                .Setup(client => client.CreateRuleManager(
+                    It.IsAny<string>(),
+                    It.IsAny<string>()))
+                .Returns(mockRuleManager.Object);
+
+            ServiceBusClient client = mockClient.Object;
+
+            // The rest of this snippet illustrates how to create and use a rule manager using the mocked service
+            // bus client above, this would be where application methods using a rule manager would be called.
+
+            string topic = "topic";
+            string subscription = "subscription";
+
+            ServiceBusRuleManager ruleManager = client.CreateRuleManager(topic, subscription);
+
+            await ruleManager.DeleteRuleAsync(RuleProperties.DefaultRuleName);
+            await ruleManager.CreateRuleAsync("filter", new CorrelationRuleFilter { Subject = "subject" });
+
+            #endregion
+        }
+
+        [Test]
+        public void TestProcessorHandlers()
```

Comment:
I think we will also need samples for the processor similar to https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample08_MockingClientTypes.md#simulating-the-eventprocessorclient-running. This can always be a separate work item as well.

Reply:
Nevermind! I saw you already mentioned this in Teams chat.